commands/run.sh

```diff
@@ -241,15 +241,35 @@ fi
 
 # Start up service dependencies in a different header to keep the main run with less noise
 if [[ "$(plugin_read_config DEPENDENCIES "true")" == "true" ]] ; then
+  exitcode=0
+
   echo "~~~ :docker: Starting dependencies"
   if [[ ${#up_params[@]} -gt 0 ]] ; then
-    run_docker_compose "${up_params[@]}" up -d --scale "${run_service}=0" "${run_service}"
+    if ! run_docker_compose "${up_params[@]}" up -d --scale "${run_service}=0" "${run_service}"; then
+      exitcode=1
+    fi
   else
-    run_docker_compose up -d --scale "${run_service}=0" "${run_service}"
+    if ! run_docker_compose up -d --scale "${run_service}=0" "${run_service}"; then
+      exitcode=1
+    fi
   fi
 
   # Sometimes docker-compose leaves unfinished ansi codes
   echo
+
+  if [[ $exitcode -ne 0 ]] ; then
+    # Dependent services failed to start.
+    echo "^^^ +++"
+    echo "+++ 🚨 Failed to start dependencies"
+
+    if [[ -n "${BUILDKITE_AGENT_ACCESS_TOKEN:-}" ]] ; then
+      print_failed_container_information
+
+      upload_container_logs "$run_service"
+    fi
+
+    return $exitcode
+  fi
 fi
```

Comment:
I've moved the if-block outside of the dependencies if block. 

As for indenting the rest of the 'run' code; I'm not sure if it would necessarily improve clarity. Additionally the current code layout does allow for easier changes to the code flow if there is a need to add some extra handling when the dependencies fail to start.